src/main/scala/com/cluster/graph/ClusterShardingApp.scala

```diff
@@ -168,6 +175,33 @@ object ClusterShardingApp {
     assert(totalMainsAckd == nMains)
     gcRef ! GlobalCoordinator.BEGIN()
     // TODO at beginning send, BEGIN(0)
+
+    // Wait until finished
+    
+
+    var finalVals: Map[Int, Option[Colour]] = null
+    while(null == finalVals){
+      Thread.sleep(1000)
+
+      val timeout: Timeout = 5.seconds
+      val sched = entityManager.scheduler
+      val future: Future[GlobalCoordinator.FinalValuesResponse] = gcRef.ask(ref => GlobalCoordinator.GetFinalValues(ref))(timeout,sched)
```

Comment:
Ahh. I assume this was the source of the bug? We were trying to access the final value map before all the data was actually received?
Good fix -- it should be noted that a valueMap may be quite large: M/Billions of vertices? -- another solution would be that each PartitionCoordinator persists their (final) value maps to their local machine. But this is good for now!

Reply:
Actually the bug is the JsonDeserialize issue mentioned below. This is just to get to the map from the GC to the main app.

And yeah, I agree this is not necessarily a good long term solution - just to get something working!